proj3_chess/chess/chess_pieces.py

```diff
@@ -0,0 +1,151 @@
+from board import Board
+from piece_properties import Player
+from abc import ABC, abstractmethod
+from coordinates import Move, Spot, MoveVector
+from typing import Iterable, Optional
+
+RIGHT = MoveVector(x=1, y=0)
+LEFT = MoveVector(x=-1, y=0)
+UP = MoveVector(x=0, y=1)
+DOWN = MoveVector(x=0, y=-1)
+UP_RIGHT = MoveVector(x=1, y=1)
+UP_LEFT = MoveVector(x=-1, y=1)
+DOWN_RIGHT = MoveVector(x=1, y=-1)
+DOWN_LEFT = MoveVector(x=-1, y=-1)
+
+
+class ChessPiece(ABC):
+    PIECE_NAME = ""
+
+    def __init__(self, player: Player) -> None:
+        self.player = player
+        self.direction = player.direction.value
+        self.symbol = player.symbol_transformation(self.PIECE_NAME)
+
+    def add_to_board(self, board: Board, spot: Spot) -> None:
+        self.board = board
```

Comment:
You're creating an instance member `board` that didn't exist before, which is a bad practice (We prefer all members be defined in the constructor, even if they are `None` at first) . You can create it in the constructor as `None`, and add a check everywhere you use it that throus an indicative error if it is `None`!!